src/domain/queues/queues-repository.ts

```diff
@@ -1,18 +1,45 @@
+import { IConfigurationService } from '@/config/configuration.service.interface';
 import { IQueuesApiService } from '@/datasources/queues/queues-api.service.interface';
 import { IQueuesRepository } from '@/domain/queues/queues-repository.interface';
-import { Inject, Injectable } from '@nestjs/common';
+import { ILoggingService, LoggingService } from '@/logging/logging.interface';
+import { WebHookSchema } from '@/routes/cache-hooks/entities/schemas/web-hook.schema';
+import { Event } from '@/routes/cache-hooks/entities/event.entity';
+import { Inject, Injectable, Scope } from '@nestjs/common';
 import { ConsumeMessage } from 'amqplib';
 
-@Injectable()
+// Only allow one instance across the app to prevent multiple queue subscriptions
+@Injectable({ scope: Scope.DEFAULT })
 export class QueuesRepository implements IQueuesRepository {
+  private readonly queueName: string;
+  private readonly listeners: Array<(event: Event) => Promise<unknown>> = [];
+
   constructor(
     @Inject(IQueuesApiService) private readonly queuesApi: IQueuesApiService,
-  ) {}
+    @Inject(IConfigurationService)
+    private readonly configurationService: IConfigurationService,
+    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+  ) {
+    this.queueName = this.configurationService.getOrThrow<string>('amqp.queue');
+  }
+
+  onModuleInit(): Promise<void> {
```

Comment:
I think we might face a race condition here: if this `onModuleInit` function is executed before `CacheHooksService.onModuleinit`, then `this.listeners` would be empty and therefore the first events wouldn't be processed by any listener, right? I'm not sure the would be lost as they wouldn't be ACKed but maybe the events processing order is modified. I'd avoid relying on the classes instantation order, but maybe I'm missing something.

What do you think about moving the complete `CacheHooksService` implementation to the domain layer instead (including its `onEvent` function). To me, it was always a bit weird that this core logic live in the routes layer so this might be a good opportunity to move it to its own class in the domain.

Reply:
Good points, with which I agree with. As of 6271732a3782d10e70f6f461531e2b22ffdfccd6, I've moved the relevant `CacheHooksService` logic to the domain. Please let me know what you think.